File: Data Structures & Algorithms/anagram-groups/submission-1.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        li=[]
        d={}
        for s in strs:
            x="".join(sorted(s))
            if x not in d:
                d[x]=[s]
            else:
                d[x].append(s)
        res=list(d.values())
        return res

File: Data Structures & Algorithms/longest-common-prefix/submission-1.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String str="";
        for(int i=0;i<strs[0].length();i++){
            for(int j=0;j<strs.length;j++){
                if (i==strs[j].length() || strs[j].charAt(i)!=strs[0].charAt(i)){
                    return str;
                }
            }
            str+=strs[0].charAt(i);
        }
        return str;
    }
}